addons/bus/models/bus_listener_mixin.py

```diff
@@ -15,12 +15,15 @@ class BusListenerMixin(models.AbstractModel):
     def _bus_send(self, notification_type, message, /, *, subchannel=None):
         """Send a notification to the webclient."""
         for record in self:
-            main_channel = record._bus_channel()
+            main_channel = record
+            while (new_main_channel := main_channel._bus_channel()) != main_channel:
+                main_channel = new_main_channel
             assert isinstance(main_channel, models.Model)
-            main_channel.ensure_one()
-            channel = main_channel if subchannel is None else (main_channel, subchannel)
-            self.env["bus.bus"]._sendone(channel, notification_type, message)
+            if main_channel:
```

Comment:
Note: the main use case of when this will be necessary: sending on a channel member of partner_id when it has no main user should simply do nothing.